### DISCUSSION TITLE
Looks like there's something wrong with this object ARCore error

### DISCUSSION THREAD

**Discussion by faizan1990** [google/model-viewer#2168] [Q&A]
This is my code:

```
<!-- Import the component -->
<script type="module" src="https://unpkg.com/@google/model-viewer/dist/model-viewer.js"></script>
<script nomodule src="https://unpkg.com/@google/model-viewer/dist/model-viewer-legacy.js"></script>
<!-- Use it like any other HTML element -->
<model-viewer src="/examples/assets/Astronaut.glb" ar camera-controls alt="A 3D model of an astronaut" background-color="#222" ios-src="assets/Astronaut.usdz" magic-leap unstable-webxr></model-viewer>
```
Serving it over HTTPS node server running locally. However, when I tap on the AR button in the Chrome browser to open the ARCore in Android Pie, I get an error saying:

**Couldn't load object: Looks like there's something wrong with this object**

Please see the attached images to see what I mean.

Any help will be really appreciated.

![Screenshot_20190524_201155_com android chrome](https://user-images.githubusercontent.com/13532645/58321390-5e44b780-7e61-11e9-8308-78d58c3fc537.jpg)

![Screenshot_20190524_201203_com google ar core](https://user-images.githubusercontent.com/13532645/58321398-64d32f00-7e61-11e9-8225-bdbb830614cf.jpg)

### Browser
Chrome

### OS
- [X] Android Pie 9.1

**Comment by cdata** [comment]
Hi @faizan1990 I'll try to confirm this but my guess is that you need a valid certificate for the origin to be considered secure. In other words, your browser should show a green lock in the location bar, not a red ⚠️

**Comment by faizan1990** [comment]
Hi @cdata you are correct! I ran it on ngrok as a valid secure server and it works fine. Thanks for your prompt reply.
